### PR TITLE
Update MySQLConnector to 2.3.x fix

### PR DESCRIPTION
**Pull request by RioMcBoo**
class SQLResult was changed

The problem is:
After Update to 2.3.x the DataReader can be automatically used for another process if first call
DataReader.Read() return "false" (due to some optimization).

We should directly set _reader = null in this case.

### CODE REVIEW

File: Source/Framework/Database/SQLResult.cs [hunk 1]
```diff
@@ -16,12 +16,17 @@ public class SQLResult
         public SQLResult(MySqlDataReader reader)
         {
             _reader = reader;
-            NextRow();
+            if (!NextRow())
```
Comment (RioMcBoo) [inline Source/Framework/Database/SQLResult.cs]: unnecessary twice check 
just use NextRow();

File: Source/Framework/Database/SQLResult.cs [hunk 2]
```diff
@@ -16,12 +16,17 @@ public class SQLResult
...
+                Close();
         }
 
-        ~SQLResult()
+        public void Close()
```
Comment (RioMcBoo) [inline Source/Framework/Database/SQLResult.cs]: Close(); never will be used externally

**Comment by ZaDarkSide** [comment]
You are missing a change in `Source/Framework/Framework.csproj`

```diff
   <ItemGroup>
     <PackageReference Include="Google.Protobuf" Version="3.25.1" />
-    <PackageReference Include="MySqlConnector" Version="2.3.1" />
+    <PackageReference Include="MySqlConnector" Version="2.3.5" />
   </ItemGroup>
   
 </Project>
 ```

**Comment by RioMcBoo** [comment]
I don't know, may be should make new issue, I fixed the bug, not updated the lib. I had done enough mistakes :)

**Comment by ZaDarkSide** [comment]
They have updated also https://github.com/mysql-net/MySqlConnector/releases/tag/2.3.5 latest is v2.3.5

**Pull request closed without merging**